app/models/category.rb

```diff
@@ -0,0 +1,5 @@
+class Category < ApplicationRecord
+  validates :name, presence: true
+
+  has_and_belongs_to_many :recipes, join_table: 'categories_recipes', foreign_key: true
```

Comment:
I don't think  you need to specify the `join_table` or `foreign_key` here. I think it's actually breaking things.
```suggestion
  has_and_belongs_to_many :recipes
```